chore: upgrade to V21 and fix Lit 2 usage and TS event names (#461)
vaadin/docs: articles/guide/upgrading.asciidoc

. Run `mvn clean install`

== Changes in Vaadin 21

These instructions apply when upgrading from a version before Vaadin 21.

> === Lit Templates are using Lit 2

Reviewer: Change to title case? Lit Templates Are Using Lit 2, as reported by Vale?

Previously, Lit templates were bases on LitElement 2.x and lit-html 1.x.

> The new Lit 2.0 is used for Lit templates in the latest Vaadin version.

Reviewer 2: ```suggestion
New Lit 2.0 is used for Lit templates in the latest Vaadin version.
```

Some changes are required to be able to use your existing template with Lit 2.0.
Refer to the https://lit.dev/docs/releases/upgrade/#update-packages-and-import-paths[Upgrade guide]
to find all necessary changes.

> In most cases you should be aware of changes in the imports:
> 
> from
> 
> [source,typescript]
> ----
> import { html, LitElement } from 'lit-element';
> ----
> 
> to
> 
> [source,typescript]
> ----
> import { html, LitElement } from 'lit';
> ----
> 
> decorator imports
> 
> from
> 
> [source,typescript]
> ----
> import { customElement, property } from 'lit-element';
> ----
> 
> to
> 
> [source,typescript]
> ----
> import { customElement, property } from 'lit/decorators.js';
> ----
> 
> and directive imports if you are using them:
> 
> from
> 
> [source,typescript]
> ----
> import { repeat } from 'lit-html/directives/repeat.js';
> ----
> 
> to
> 
> [source,typescript]
> ----
> import { repeat } from 'lit/directives/repeat.js';
> ----

Reviewer 2: ```suggestion
Most of the necessary changes are with imports.

You need to change imports of `html` and [classname]#LitElement# from:

[source,typescript]
----
import { html, LitElement } from 'lit-element';
----

to:

[source,typescript]
----
import { html, LitElement } from 'lit';
----

Change `customElement` and `property` decorator imports from:

[source,typescript]
----
import { customElement, property } from 'lit-element';
----

to

[source,typescript]
----
import { customElement, property } from 'lit/decorators.js';
----

If you are using `repeat` directive imports, change them from:

[source,typescript]
----
import { repeat } from 'lit-html/directives/repeat.js';
----

to

[source,typescript]
----
import { repeat } from 'lit/directives/repeat.js';
----
```

Reviewer 3: These imports are just examples of different type of imports and how they should be changed. This is not an exhaustive list so mentioning only "Change `customElement` and `property` decorator imports" or "If you are using `repeat` directive imports" explicitly is not good.

I'll use your suggestion as a basis but make some changes to the text content to make it more clear.

Reviewer 2: OK, the main point was that the text was formatted like it was a really long sentence with code examples in between.

Reviewer 3: Updated.

Some Lit APIs have been renamed.
The most noticeable change is that the `@internalProperty` decorator has been renamed to `@state`.

== Changes in Vaadin 20

> These instructions apply when upgrading from a version before Vaadin 20.

Reviewer 2: ```suggestion
The following instructions apply when upgrading from a version before Vaadin 20.
```

== Changes in Vaadin 21

The following instructions apply when upgrading from a version before Vaadin 21.

=== Lit Templates Use Lit 2

Previously, Lit templates were based on LitElement 2.x and lit-html 1.x.
New Lit 2.0 is used for Lit templates in the latest Vaadin version.
Some changes are required to be able to use your existing template with Lit 2.0.
Please refer to the https://lit.dev/docs/releases/upgrade/#update-packages-and-import-paths[Lit Upgrade Guide] to find all the necessary changes.
Most of the necessary changes are with imports.

Use the `lit` package.

> For example change imports of `html` and [classname]#LitElement# from:

Reviewer 2: ```suggestion
For example, change imports of `html` and [classname]#LitElement# from:
```

[source,typescript]
----
import { html, LitElement } from 'lit-element';
----

to:

[source,typescript]
----
import { html, LitElement } from 'lit';
----

Update decorator imports.
Decorators have been moved to a separate module.

> For example change `customElement` and `property` decorator imports from:

Reviewer 2: ```suggestion
For example, change `customElement` and `property` decorator imports from:
```
